ethminer/main.cpp

```diff
@@ -61,6 +65,26 @@ int main(int argc, char** argv)
 	setenv("GPU_MAX_ALLOC_PERCENT", "100");
 	setenv("GPU_SINGLE_ALLOC_PERCENT", "100");
 
+#if defined(_WIN32)
+    // Set output mode to handle virtual terminal sequences
+	HANDLE hOut = GetStdHandle(STD_OUTPUT_HANDLE);
+	if (hOut == INVALID_HANDLE_VALUE)
+	{
+		return GetLastError();
+	}
+	
+	DWORD dwMode = 0;
+	if (!GetConsoleMode(hOut, &dwMode))
+	{
+		return GetLastError();
+	}
+	dwMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+	if (!SetConsoleMode(hOut, dwMode))

```

Comment:
Wouldn't this fail on anything but Win 10?

Reply:
In addition, this should be reverted on exit, no?

Reply:
> Wouldn't this fail on anything but Win 10?

Yes, this is correct, only works with win 10, maybe we should change the "return" to something else. Also a non win10 variant should remove the escape codes otherwise see screenshot.

> In addition, this should be reverted on exit, no?

as far as i can test, this is only for the current std output. i did a test with a version which sets the console mode and a variant afterwards which does not set it.

![colors2](https://user-images.githubusercontent.com/805790/35795212-c32d05c2-0a58-11e8-9c72-b87757180f3d.png)

Reply:
Looked so easy with ENABLE_VIRTUAL_TERMINAL_PROCESSING :)

Maybe it is a better solution cleaning up #226 ?

Reply:
This is simple and elegant. Why not just test for Win 10 and only do it then. I think most of the Windows reports here are on 10 anyway. You'd need a separate set of empty escapes for non-win10.

Reply:
I don't know how clean it would be to get rig of the escape stuff, since the ``EthReset`` etc. are defines. Only way i can think of is using a regex (i have one) to remove all colors on the fly on non win10 systems.

Reply:
They don't have to be defines! But I understand, no longer the simple elegant solution.

Reply:
@smurfy Could merge it and see... Don't imagine there'd be much of an outcry. Who's not running win 10 these days?

Reply:
@smurfy Fix the exit on fail for non win 10 prior to merging